src/test/java/com/google/devtools/build/lib/starlark/StarlarkIntegrationTest.java

```diff
@@ -723,6 +723,66 @@ public void testDefaultInfoWithRunfilesConstructor() throws Exception {
     assertThat(getConfiguredTarget("//src:r_tools")).isNotNull();
   }
 
+  private void helperDefaultInfoFilesAddedToNativeBinaryTargetRunfiles(String rule,
+      String... extraAttrs) throws Exception {
+    scratch.file(
+        "test/starlark/extension.bzl",
+        "def custom_rule_impl(ctx):",
+        "  out = ctx.actions.declare_file(ctx.attr.name + '.out')",
+        "  ctx.actions.write(out, 'foobar')",
+        "  return [DefaultInfo(files = depset([out]))]",
+        "",
+        "custom_rule = rule(implementation = custom_rule_impl)");
+
+    scratch.file(
+        "test/starlark/BUILD",
+        "load('//test/starlark:extension.bzl', 'custom_rule')",
+        "",
+        "custom_rule(name = 'cr')",
+        String.format("%s(name = 'binary', data = [':cr'], %s)", rule,
+            String.join(", ", extraAttrs))
+    );
+
+    ConfiguredTarget target = getConfiguredTarget("//test/starlark:binary");
+
+    assertThat(target.getLabel().toString()).isEqualTo("//test/starlark:binary");
+    assertThat(
+        ActionsTestUtil.baseArtifactNames(
+            target.getProvider(RunfilesProvider.class).getDefaultRunfiles().getAllArtifacts()))
+        .contains("cr.out");
+    assertThat(
+        ActionsTestUtil.baseArtifactNames(
+            target.getProvider(RunfilesProvider.class).getDataRunfiles().getAllArtifacts()))
+        .contains("cr.out");
+  }
+
+  @Test
+  public void testDefaultInfoFilesAddedToCcBinaryTargetRunfiles() throws Exception {
+    helperDefaultInfoFilesAddedToNativeBinaryTargetRunfiles("cc_binary");
```

Comment:
Readability and testing best practices: 
Please inline `helperDefaultInfoFilesAddedToNativeBinaryTargetRunfiles` in all 3 tests. 
This helps debugging when the test breaks.

Reply:
Done. I will keep this in mind for future contributions.